deepnote_toolkit/runtime_patches.py

```
@@ -1,10 +1,64 @@
+        from trino import client as trino_client
+
+        _original_execute = trino_client.TrinoQuery.execute
+        _original_fetch = trino_client.TrinoQuery.fetch
+
+        def _cancel_on_error(query: "trino_client.TrinoQuery") -> None:
+            """Best-effort cancel when an error occurs."""
+            if not query._cancelled and query._next_uri:
+                try:
+                    query.cancel()
+                except (KeyboardInterrupt, Exception):
+                    pass
```

Ruff (0.14.14)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)